### ISSUE TITLE
bib report missing class name

### ISSUE THREAD

**Issue by jianyangli** [NASA-Planetary-Science/sbpy#179]
When add a bib entry to a class method, the class name is missing in the bib report.  For example, execute below code piece:

```
from sbpy import bib
from sbpy.bib import cite

class A():
    @cite({'ref': '1989aste.conf..524B'})
    def test(self):
        pass

bib.track()
a = A()
a.test()
print(bib.to_text())
```

It will print:

```
__main__.test:
  ref:
      Bowell, Hapke, Domingue et al. 1989, Asteroids II, 524
```

The class name `A` is missing in the report `__main__.test:`.  Can we add class name?

**Comment by jianyangli** [comment]
Just discovered that a related problem is that the report will merge all entries with the same function name, despite whether they are from the same class or different classes:

```
from sbpy import bib
from sbpy.bib import cite

class A():
    @cite({'ref': '1989aste.conf..524B'})
    def test(self):
        pass

class B():
    @cite({'ref': '2010Icar..209..542M'})
    def test(self):
        pass

class C():
    @cite({'ref': '2016P&SS..123..117P'})
    def test(self):
        pass

bib.track()
a = A()
a.test()
b = B()
b.test()
c = C()
c.test()

print(bib.to_text())
```

will produce output:

```
__main__.test:
  ref:
      Bowell, Hapke, Domingue et al. 1989, Asteroids II, 524
      Penttilä, Shevchenko, Wilkman & Muinonen 2016, Planetary and Space Science, Vol 123, 117
      Muinonen, Belskaya, Cellino et al. 2010, Icarus, Vol 209, 2, 542
```

**Comment by mkelley** [comment]
Both cases were fixed with PR #190 .
